src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1593,6 +1598,26 @@ private static PropertySymbol GetPropertySymbol(BoundExpression expr, out BoundE
             return propertySymbol;
         }
 
+#nullable enable
+        internal static Symbol? GetIndexerOrImplicitIndexerSymbol(BoundExpression? e)
```

Comment:
>GetIndexerOrImplicitIndexerSymbol [](http://example.com/codeflow?start=32&length=33)

It doesn't look like this method handles all inputs that BoundListPattern/BoundSlicePattern can use in place of IndexerAccess. I think this is the root cause of the IOperation leg failure.